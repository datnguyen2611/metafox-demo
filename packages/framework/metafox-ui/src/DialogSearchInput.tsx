import { SearchBox } from '@metafox/ui';
import { InputBaseProps } from '@mui/material';
import React from 'react';

export interface SearchInputProps extends InputBaseProps {
  onChanged: (value: string) => void;
}

export default function DialogSearchInput({
  placeholder,
  onChanged,
  ...rest
}: SearchInputProps) {
  return (
    <SearchBox
      data-testid="searchBox"
      placeholder={placeholder}
      onChange={evt => onChanged(evt.currentTarget.value)}
      {...rest}
    />
  );
}
