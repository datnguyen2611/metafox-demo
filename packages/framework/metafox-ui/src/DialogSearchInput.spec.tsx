import { render } from '@testing-library/react';
import * as React from 'react';
import DialogSearchInput from './DialogSearchInput';

describe('DialogSearchInput', () => {
  it('render', async () => {
    const props = {
      placeholder: 'typing key search....'
    };
    const { container } = render(<DialogSearchInput {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('render onchange', async () => {
    const props = {
      placeholder: 'typing key search....',
      onChanged: () => jest.fn()
    };

    const utils = render(<DialogSearchInput {...props} />);
    const input = utils.getByTestId('searchBox');

    expect(input).toBeInTheDocument();

    expect(utils.container).toMatchSnapshot();
  });
});
