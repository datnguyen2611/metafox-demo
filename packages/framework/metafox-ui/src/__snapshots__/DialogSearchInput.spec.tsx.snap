// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogSearchInput render 1`] = `
<div>
  <div
    aria-controls="search_complete"
    class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedStart css-1fngovj-MuiInputBase-root-InputBaseStyled-root"
    variant="search"
  >
    <span
      class="ico ico-search-o css-1o2x2pn-MuiIcon-root"
    />
    <input
      aria-autocomplete="list"
      aria-expanded="true"
      aria-label="typing key search...."
      autocomplete="off"
      class="MuiInputBase-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedStart css-19fibzo-MuiInputBase-input"
      data-testid="searchBox"
      name="search"
      placeholder="typing key search...."
      role="combobox"
      spellcheck="false"
      type="search"
      value=""
    />
  </div>
</div>
`;

exports[`DialogSearchInput render onchange 1`] = `
<div>
  <div
    aria-controls="search_complete"
    class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedStart css-1fngovj-MuiInputBase-root-InputBaseStyled-root"
    variant="search"
  >
    <span
      class="ico ico-search-o css-1o2x2pn-MuiIcon-root"
    />
    <input
      aria-autocomplete="list"
      aria-expanded="true"
      aria-label="typing key search...."
      autocomplete="off"
      class="MuiInputBase-input MuiInputBase-inputTypeSearch MuiInputBase-inputAdornedStart css-19fibzo-MuiInputBase-input"
      data-testid="searchBox"
      name="search"
      placeholder="typing key search...."
      role="combobox"
      spellcheck="false"
      type="search"
      value=""
    />
  </div>
</div>
`;
